Runtime property being resolved at build time

### Describe the bug

Some configuration properties which are supposed to be resolved at runtime are being resolved at build time, resulting in build errors when referencing runtime properties such as `user.home`. This appears to be a regression on main and does not affect the released version 3.15.1.

### Expected behavior

`quarkus.log.file.path` is resolved at runtime.

### Actual behavior

`quarkus.log.file.path` is resolved at build time.

### How to Reproduce?

Reproducer: [quarkus-reproducer.zip](https://github.com/user-attachments/files/17595956/quarkus-reproducer.zip)

**Steps to reproduce**

Add or edit this line in `application.properties`:

```
quarkus.log.file.path=${user.home}/quarkus-reproducer.log
```

Run `./gradlew quarkusAppPartsBuild`.

The following error is reported:

```
> Task :quarkusAppPartsBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusAppPartsBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > Configuration validation failed:
                java.util.NoSuchElementException: SRCFG00011: Could not expand value user.home in property quarkus.log.file.path
```

### Output of `uname -a` or `ver`

Darwin Rods-MacBook-Pro.local 24.0.0 Darwin Kernel Version 24.0.0: Tue Sep 24 23:35:10 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T6031 arm64

### Output of `java -version`

openjdk version "21.0.4" 2024-07-16 OpenJDK Runtime Environment Homebrew (build 21.0.4) OpenJDK 64-Bit Server VM Homebrew (build 21.0.4, mixed mode, sharing)

### Quarkus version or git rev

e81fa0c828aab38ffb50a0adf61aacb6d376c43c

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.9

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java
@@ -84,6 +84,7 @@ import io.smallrye.config.PropertyName;
 import io.smallrye.config.SecretKeys;
 import io.smallrye.config.SmallRyeConfig;
 import io.smallrye.config.SmallRyeConfigBuilder;
+import io.smallrye.config.SmallRyeConfigBuilderCustomizer;
 import io.smallrye.config.SysPropConfigSource;
 import io.smallrye.config.common.AbstractConfigSource;
 
@@ -1112,6 +1113,17 @@ public final class BuildTimeConfigurationReader {
             builder.getProfiles().add("");
             builder.getSources().clear();
             builder.getSourceProviders().clear();
+            builder.withCustomizers(new SmallRyeConfigBuilderCustomizer() {
+                @Override
+                public void configBuilder(final SmallRyeConfigBuilder builder) {
+                    builder.getMappingsBuilder().getMappings().clear();
+                }
+
+                @Override
+                public int priority() {
+                    return Integer.MAX_VALUE;
+                }
+            });
             builder.setAddDefaultSources(false)
                     // Customizers may duplicate sources, but not much we can do about it, we need to run them
                     .addDiscoveredCustomizers()

